eng/packages/http-client-csharp-mgmt/generator/Azure.Generator.Management/src/Providers/MockableResourceProvider.cs

```diff
@@ -0,0 +1,151 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Core;
+using Azure.Generator.Management.Snippets;
+using Azure.Generator.Management.Utilities;
+using Azure.ResourceManager;
+using Humanizer;
+using Microsoft.TypeSpec.Generator.Primitives;
+using Microsoft.TypeSpec.Generator.Providers;
+using Microsoft.TypeSpec.Generator.Statements;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using static Microsoft.TypeSpec.Generator.Snippets.Snippet;
+
+namespace Azure.Generator.Management.Providers
+{
+    internal class MockableResourceProvider : TypeProvider
+    {
+        private protected readonly IReadOnlyList<ResourceClientProvider> _resources;
+
+        private const string IdProperty = "Id";
```

Comment:
```suggestion
        private const string IdPropertyName = "Id";
```

Reply:
well I think I should remove this and replace it by snippets